test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,124 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/daemonset"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/deployment"
+)
+
+var _ = Describe("NMState operator", func() {
+	Context("when installed for the first time", func() {
+		BeforeEach(func() {
+			installDefaultNMState()
+		})
+		It("should deploy daemonset and webhook deployment", func() {
+			daemonset.Eventually(getDefaultNMStateHandlerKey()).Should(daemonset.BeReady())
+			deployment.Eventually(getDefaultNMStateWebhookKey()).Should(deployment.BeReady())
+		})
+		AfterEach(func() {
+			uninstallDefaultNMState()
+		})
+	})
+	Context("when NMState is installed", func() {
+		BeforeEach(func() {
+			installDefaultNMState()
+		})
+		It("should list one NMState CR", func() {
+			ds, err := daemonset.GetDaemonSetList(getDefaultHandlerLabels())
+			Expect(err).ToNot(HaveOccurred(), "List daemon sets in namespace nmstate succeeds")
+			Expect(ds.Items).To(HaveLen(1), "and has only one item")
+		})
+		Context("and the CR is created with a wrong name", func() {
+			var nmstate = getDefaultNMState()
+			nmstate.Name = "wrong-name"
+			BeforeEach(func() {
+				err := framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+				Expect(err).ToNot(HaveOccurred(), "NMState CR with incorrect name is created without error")
+			})
+			AfterEach(func() {
+				err := framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred(), "NMState CR with incorrect name is removed without error")
+			})
+			It("should ignore it", func() {
+				ds, err := daemonset.GetDaemonSetList(getDefaultHandlerLabels())
+				Expect(err).ToNot(HaveOccurred(), "Daemon set list is retreieved without error")
+				Expect(ds.Items).To(HaveLen(1), "and still only has one item")
+			})
+		})
+		AfterEach(func() {
+			uninstallDefaultNMState()
+		})
+	})
+	Context("when NMState is uninstalled", func() {
```

Comment:
If you move this `Context` under `Context("when NMState is installed", func() {` rename this as 

```golang
Context("and uninstalled", func() {
```

Then you don't need to install here again and Ginkgo will compose exactly what is doing at the test name.